File: app/src/main/java/com/runmyrobot/android_robot_for_phone/utils/SabertoothDriverUtil.java
package com.runmyrobot.android_robot_for_phone.utils;

/**
 * Created by Brendon on 9/7/2018.
 */
public class SabertoothDriverUtil {
    public static final byte STOP_BYTE = 0;

    public SabertoothDriverUtil() {
    }

    public static byte getDriveSpeed(byte driveSpeed, int motorNum) {
        switch(motorNum) {
            case 0:
                return (byte)((int) ValueUtil.map((float)driveSpeed, -128.0F, 127.0F, 1.0F, 127.0F, 1.0F));
            case 1:
                return (byte)((int) ValueUtil.map((float)driveSpeed, -128.0F, 127.0F, -128.0F, -1.0F, 1.0F));
            default:
                return (byte)((int) ValueUtil.map(0.0F, -128.0F, 127.0F, 1.0F, 127.0F, 1.0F));
        }
    }
}

File: app/src/main/java/com/runmyrobot/android_robot_for_phone/utils/SabertoothDriverUtil.kt
package com.runmyrobot.android_robot_for_phone.utils

/**
 * Created by Brendon on 9/7/2018.
 */
class SabertoothDriverUtil {
    companion object {
        val STOP_BYTE: Byte = 0

        fun getDriveSpeed(driveSpeed: Byte, motorNum: Int, scale : Float = 1f): Byte {
            return when (motorNum) {
                0 -> ValueUtil.map(driveSpeed.toFloat(), -128.0f, 127.0f, 1.0f, 127.0f, scale).toInt().toByte()
                1 -> ValueUtil.map(driveSpeed.toFloat(), -128.0f, 127.0f, -128.0f, -1.0f, scale).toInt().toByte()
                else -> ValueUtil.map(0.0f, -128.0f, 127.0f, 1.0f, 127.0f, scale).toInt().toByte()
            }
        }
    }
}
